{\bf [20 pts] Haplotyping by hand}

This will be a simple exercise in haplotyping that should be done by hand. Please show your work.

After mapping reads and performing variant calling, you have filtered down to only heterozygous sites and encoded the variants in 0/1 format to denote wild-type/variant.

Below each read is listed on a line, where `-' means that the read doesn't cover that variant locus.
\begin{verbatim}
----------------1101011111-----------------------------
----------10100100101000010011110----------------------
-----001000101101101-----------------------------------
0011000100010------------------------------------------
------101110100100101----------------------------------
------------------------------001010001010101----------
------------------------------1101011101010------------
----------------------------000010100010---------------
----------------------00000011110101-------------------
-----------------------------------------01010001010---
---------------------------------------------1110101011
-------------------------------------------010001110100
-----------------------------------00010101010---------
1100111010101------------------------------------------
----------------001010000000---------------------------
----------------------1111110010101--------------------
-------------------------------------10101010111010----
----------01011011010----------------------------------
\end{verbatim}

\begin{enumerate}[label=(\alph*)]
    \item (10pts) Assuming the reads come from a diploid organism, determine the two haplotypes.
    \begin{solution}

    \end{solution}
    \item (5pts) What is the MEC score of your proposed diploid haplotypes?
    \begin{solution}

    \end{solution}
    \item (5pts) You should have noticed that in your solution, the two haplotypes are entirely complementary (if there is a 0 in haplotype A, then there is a 1 in haplotype B). Explain why this is the case.
    \begin{solution}

    \end{solution}
\end{enumerate}