{\bf [40 pts] Metagenomic read binning}

In this problem, we will be building a simple metagenomic read classifer from scratch.

\begin{enumerate}[label=(\alph*)]
    \item (10pts)  Inside the data folder is 4 genome fasta files, each of which are 1,000,000 bp long. Using $k=4$, report the 10 most common kmers for each genome.
    \begin{solution}


    \end{solution}
    \item (10pts) The reads.fa file contains 1000 reads from the synthetic genomes. Using the most common kmers from each genome, design a classifier to determine the source genome of the reads. Describe your method below, and implement it with code. State the relative frequency of each genome in the reads file.
    \begin{solution}

    \end{solution}   

    \item (10pts) An alternative method for binning metagenomic reads is to match them to discriminative kmers. A kmer will be discriminative if it is present in only 1 of the reference genomes. Using k = 10, report the 5 most common discriminative kmers for each genome.
    \begin{solution}

    \end{solution}   

    \item (10pts) Similar to part b), design a new classifier, this time using discriminative kmers from each genome. Describe your method below, and then state the relative frequency of each genome in the reads file. (Hint: for this method, would recommend using the entire set of discriminative kmers for a given genome)
    \begin{solution}

    \end{solution}   

\end{enumerate}





